ze/btx_zeinterval_callbacks.cpp

```diff
@@ -765,87 +765,281 @@ static void zeEventDestroy_exit_callback(void *btx_handle, void *usr_data, int64
  * Sampling
  */
 
-static void lttng_ust_ze_sampling_gpu_energy_callback(void *btx_handle, void *usr_data, int64_t ts,
-                                                      const char *hostname, int64_t vpid,
-                                                      uint64_t vtid, ze_device_handle_t hDevice,
-                                                      uint32_t domain, uint64_t energy,
-                                                      uint64_t sampling_ts) {
-
+DeviceHash get_device_hash(void *usr_data, const char *hostname, int64_t vpid,
+                           ze_device_handle_t hDevice) {
   auto *data = static_cast<data_t *>(usr_data);
-  auto [it, inserted] = data->device_energy_ref.insert(
-      {{hostname, vpid, hDevice, domain}, {energy, sampling_ts, ts}});
-  // First entry
-  if (inserted)
-    return;
-  auto &[prev_energy, prev_sampling_ts, prev_ts] = it->second;
+  const auto it0 = data->sampling_device_property.find({hostname, vpid, hDevice});
+  if (it0 != data->sampling_device_property.cend()) {
+    const auto &[deviceProp, deviceIdx] = it0->second;
+
+    uint64_t hash = 0xcbf29ce484222325; // FNV offset basis
+    for (int i = 0; i < ZE_MAX_DEVICE_UUID_SIZE; i++) {
+      hash ^= (uint64_t)deviceProp.core.uuid.id[i];
+      hash *= 0x100000001b3; // FNV prime
+    }
 
-  // Watt conversion
-  btx_push_message_lttng_power(btx_handle, hostname, 0, 0, prev_ts, BACKEND_ZE, (uint64_t)hDevice,
-                               (thapi_domain_idx)domain,
-                               (energy - prev_energy) / (double)(sampling_ts - prev_sampling_ts));
+    return {hash, deviceIdx};
+  }
+  return {0, 0}; // Return 0 values if not found
```

Comment:
I guess we should use optional to be cleaner.

Reply:
(I don't understand why it was  put as resolved)

Reply:
Sorry, it was mixed up. I used optional with a message to handle not found.